src/main/kotlin/com/example/iotmanagementdevice/model/MongoRole.kt

```diff
@@ -0,0 +1,29 @@
+package com.example.iotmanagementdevice.model
+
+import org.bson.types.ObjectId
+import org.springframework.data.annotation.Id
+import org.springframework.data.annotation.TypeAlias
+import org.springframework.data.mongodb.core.mapping.Document
+import org.springframework.security.core.GrantedAuthority
+
+@TypeAlias("Role")
+@Document(collection = MongoRole.COLLECTION_NAME)
+data class MongoRole(
+    @Id
+    val id: ObjectId?,
+    val roleName: RoleName?,
+) : GrantedAuthority {
+
+    override fun getAuthority(): String {
+        return "ROLE_" + roleName?.name
+    }
```

Comment:
SRP is violated here. this class is DB entity and Sping Security entity at the same time

Reply:
Switch to SimpleGrantedAuthority